pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -207,9 +208,10 @@ async def run(self):
                     logger.info("Verifying associated image advisory %s...", image_advisory)
                     image_advisory_info = await self.get_advisory_info(image_advisory)
                     try:
-                        self.verify_image_advisory(image_advisory_info)
-                        live_id = self.get_live_id(image_advisory_info)
-                        assert live_id
+                        self.verify_advisory_live_id(image_advisory_info)
+                        if assembly_type != assembly.AssemblyTypes.CANDIDATE:
+                            self.verify_advisory_status(image_advisory_info)
+                        live_id = self.get_live_id(image_advisory_info) or 'live-id'
```

Comment:
IIUC, when `live_id` is None `verify_advisory_live_id()` will throw an exception. So what situation do you expect to cover with this check?

Also, `verify_advisory_live_id()` is just running `get_live_id()` which you are doing again at line 214, then possibly throw an exception. Wouldn't it make sense to re-write this as something like:

```
live_id = self.get_live_id(image_advisory_info)
if not live_id:
    raise VerificationError(f"Advisory {advisory_info['id']} doesn't have a live ID.")
if assembly_type != assembly.AssemblyTypes.CANDIDATE:
    self.verify_advisory_status(image_advisory_info)
```

and get rid of the redundant `verify_advisory_live_id()` function?

Reply:
Right, I was thinking about `errata_url` still having something other than an empty string when live_id was None, but that doesn't happen since the exception gets raised. I think it's fine for errata_url to stay an empty string instead of a dummy value "access.redhat.com/errata/live-id" - cincinnat/ota team might be relying on it to be that way. Updated with your suggestion